demos/projects/ST/stm32h745i_discovery/cm7/main.h

```diff
@@ -0,0 +1,71 @@
+/* USER CODE BEGIN Header */
```

Comment:
I see inconsistent usage of main.h (e.g. in NXP it is checked in but not used). 
Should we get the common APIs required by the sample defined here then include in the sample c file (e.g. the Error_Handler, Random and Time PAL functions).

Reply:
ST projects are created using CubeMX and that generates main.h, just to keep it similar to ST experience I kept it same. 
Regarding NXP I am not seeing any file name main.h in main branch of this repo (https://github.com/Azure-Samples/iot-middleware-freertos-samples/tree/main/demos/projects/NXP/mimxrt1060), am I missing something here?

main.c is more attached to platform and wanted to make sure it remain same to how that platform provides its sample to boot.

Reply:
Ah, OK - a prior version or something happened to my fork - I have main.h in NXP but wasn't referenced.